opengrok-indexer/src/main/java/org/opengrok/indexer/search/SettingsHelper.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2011, 2019, Oracle and/or its affiliates. All rights reserved.
+ * Portions copyright (c) 2011 Jens Elkner.
+ * Portions Copyright (c) 2017-2019, Chris Fraire <cfraire@me.com>.
+ */
+package org.opengrok.indexer.search;
+
+import org.apache.lucene.index.IndexReader;
+import org.opengrok.indexer.configuration.Project;
+import org.opengrok.indexer.index.IndexAnalysisSettings3;
+import org.opengrok.indexer.index.IndexAnalysisSettingsAccessor;
+import org.opengrok.indexer.index.IndexedSymlink;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Represents a helper class for accessing settings.
+ * @author Jens Elkner
+ */
+public class SettingsHelper {
+
+    private final IndexReader reader;
+
+    /**
+     * Key is Project name or empty string for null Project.
+     */
+    private Map<String, IndexAnalysisSettings3> mappedAnalysisSettings;
+
+    /**
+     * Key is Project name or empty string for null Project. Map is ordered by
+     * canonical length (ASC) and then canonical value (ASC).
+     */
+    private Map<String, Map<String, IndexedSymlink>> mappedIndexedSymlinks;
+
+    public SettingsHelper(IndexReader reader) {
+        if (reader == null) {
+            throw new IllegalArgumentException("reader is null");
+        }
+        this.reader = reader;
+    }
+
+    /**
+     * Gets any mapped symlinks (after having called {@link #getSettings(String)}).
+     * @return either a defined map or {@code null}
+     */
+    public Map<String, IndexedSymlink> getSymlinks(String projectName) {
+        if (mappedIndexedSymlinks == null) {
+            throw new IllegalStateException("getSettings() not yet called");
```

Comment:
is it problem to call it here `getSettings()`?

Reply:
No doesn't seem so